updating docker build instructions and guidance

vinnnyr opened the pull request:
As per https://github.com/ros-planning/navigation2/issues/2388 Docker instructions on the documentation need to change. 
It is now recommended to install from apt, unless for development purposes.

SteveMacenski commented on build_instructions/index.rst:
```diff
@@ -147,9 +147,14 @@ The rest of this should look familiar.
 
-There are 2 options for docker with Nav2:
-building a container and using the DockerHub container.
+  To avoid using `sudo` to use docker, add your use to the `docker` group as outlined here for a Linux host machine: https://docs.docker.com/engine/install/linux-postinstall/
```
Remove the note, I'm not sure we should be recommending people to mess with their system security settings. It is easy to really mess things up in an unrecoverable way and I don't want to deal with devops tickets that people messed up their systems.

SteveMacenski replied:
Just include sudo

SteveMacenski commented on build_instructions/index.rst:
```diff
@@ -147,9 +147,14 @@ The rest of this should look familiar.
+  To avoid using `sudo` to use docker, add your use to the `docker` group as outlined here for a Linux host machine: https://docs.docker.com/engine/install/linux-postinstall/
+  If you choose not to do this, you must run the below docker commands as `sudo`
+
+The official Dockerfile and Dockerhub container are primarily for use in the Nav2 CI
```
```suggestion
The official Dockerhub entries are primarily for use in the Nav2 CI, but they may also be used for development use to get a docker image that tracks Nav2 ``main`` branch. The ``Dockerfile`` in the root of the repository is recommended for production use, set to your distribution of choice. 
```

SteveMacenski commented on build_instructions/index.rst:
```diff
@@ -147,9 +147,14 @@ The rest of this should look familiar.
+
+The official Dockerfile and Dockerhub container are primarily for use in the Nav2 CI
+
+For Nav2 production containerized use, it is reccomended to install Nav2 releases from the apt repository inside a ROS container
```
```suggestion
It is though generally recomended to install Nav2 releases from the apt repository inside a container if you'd like to use our released binaries.
```

SteveMacenski commented on build_instructions/index.rst:
```diff
@@ -159,28 +164,28 @@ Building Docker Container
 Note: You may also need to configure your docker for DNS to work. See article here for details: https://development.robinwinslow.uk/2016/06/23/fix-docker-networking-dns/
 
-.. rst-class:: content-collapse
+If you would like to build from dockerhub cache to speed up the build
```
Please don't fully remove instructions for using the dockerhub containers, they're still very useful for developers that need the main branch status of Nav2, its still a great ready to use container for that. The caching instructions are also a nice touch, but it should be both, not either or

vinnnyr commented:
Ok I think I have addressed your comments, and I think I am happy with the result https://825-240847415-gh.circle-artifacts.com/0/html/build_instructions/index.html#docker

vinnnyr commented on build_instructions/index.rst:
```diff
@@ -171,16 +172,23 @@ If proxies are needed:
 =========================
-
 We allow for you to pull the latest docker image from the main branch at any time. As new releases and tags are made, docker containers on docker hub will be versioned as well to chose from.
+This docker image will not contain a built overlay, and you must build the overlay Nav2 workspace yourself (see Build Nav2 Main up above).
```
Yes this is true, I think the circleci config changed? See Point 2 of Ruffin's response: https://github.com/ros-planning/navigation2/issues/2388#issuecomment-854324100

Further evidence is in the container itself
```
$ docker pull rosplanning/navigation2:main
main: Pulling from rosplanning/navigation2
Digest: sha256:f60e9af8d32a897362c7ba869906ffb4531f269c1f25d9222dfd896a77a193ed
Status: Image is up to date for rosplanning/navigation2:main
docker.io/rosplanning/navigation2:main

$ docker run -it rosplanning/navigation2:main
root@03340a985d2c:/opt/overlay_ws# ls
src
```

SteveMacenski replied:
Huh, ok
